extensions/interactions/LogicProof/LogicProof.js

```diff
@@ -49,22 +49,38 @@ oppia.directive('oppiaInteractiveLogicProof', [
         );
         $scope.questionData.language.operators = $scope.typing[0].operators;
 
-        $scope.assumptionsString = logicProofShared.displayExpressionArray(
-          $scope.questionData.assumptions,
-          $scope.questionData.language.operators);
-        $scope.targetString = logicProofShared.displayExpression(
-          $scope.questionData.results[0],
-          $scope.questionData.language.operators);
+        $scope.displayExpression = function(exp) {
+          return logicProofShared.displayExpression(
+            exp, $scope.questionData.language.operators);
+        };
+
+        $scope.displayExpressionArray = function(arr) {
+          return logicProofShared.displayExpressionArray(
+            arr, $scope.questionData.language.operators);
+        };
+
+        if ($scope.questionData.assumptions.length <= 1) {
+          $scope.assumptionsString = $scope.displayExpressionArray(
+            $scope.questionData.assumptions);
+        } else {
+          $scope.assumptionsString = $scope.displayExpressionArray(
+            $scope.questionData.assumptions.slice(
+              0, $scope.questionData.assumptions.length - 1)
+            ) + ' and ' + $scope.displayExpression(
+              $scope.questionData.assumptions[
+                $scope.questionData.assumptions.length - 1]);
+        }
+        $scope.targetString = $scope.displayExpression(
+          $scope.questionData.results[0]);
         $scope.questionString = ($scope.assumptionsString === '') ?
             'Prove ' + $scope.targetString + '.':
             'Assuming ' + (
               $scope.assumptionsString + '; prove ' + $scope.targetString +
               '.');
 
-
         $scope.questionInstance = logicProofStudent.buildInstance(
           $scope.questionData);
-        $scope.haveErrorMessage = false;
+        $scope.messageIsSticky = false;
```

Comment:
I don't understand what this means, from the name. Can you add an explanatory comment?